src/osemgrep/language_server/custom_requests/Search.mli

```diff
@@ -1,11 +1,28 @@
-val meth : string
+val start_meth : string
 (** method to match on: semgrep/search *)
 
 val mk_params :
   lang:Xlang.t option -> fix:string option -> string -> Jsonrpc.Structured.t
 
-val on_request :
-  RPC_server.t -> Jsonrpc.Structured.t option -> Yojson.Safe.t option
-(** [on_request runner request] will run [runner] on the given pattern and optional
-    language params. If the request is None, we return None. Otherwise, we return
-    [Some (JSON response)] that contains the ranges of all matches *)
+val ongoing_meth : string
+(** method to match on: semgrep/searchOngoing *)
+
+val start_search :
+  RPC_server.t ->
+  Jsonrpc.Structured.t option ->
+  Yojson.Safe.t option * RPC_server.t
+(** [start_search server params] will cause a search to start with the given parameters,
+    storing the information of remaining rules/targets to search in the server session.
+    It will then return the matches in the first file with matches.
+    Will return `Assoc ["locations": `List []] when the search has concluded.
+  *)
+
+val search_next_file :
+  RPC_server.t ->
+  Jsonrpc.Structured.t option ->
+  Yojson.Safe.t option * RPC_server.t
+(** [search_next_file server params] is used during an ongoing search, and will
+    return the matches in the first file with matches, based on the remaining
+    rules/targets to search in the server session state.
+    Will return `Assoc ["locations": `List []] when the search has concluded.
```

Comment:
Would be nice to use LSP spec's partial results https://github.com/microsoft/vscode-languageserver-node/blob/e1345cca06095d61d3c5c0ff961472d55cbcf798/server/src/node/test/connection.test.ts#L141-L180

Reply:
this ended up being infeasible due to latency (mass notifications sent to the extension had a six second delay with `django`). Unclear why this is happening, we will choose to proceed this way and try this again later.